src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_producer_power_formula.py

```diff
@@ -34,32 +34,26 @@ def generate(self) -> FormulaEngine[Power]:
         builder = self._get_builder(
             "producer_power", ComponentMetricId.ACTIVE_POWER, Power.from_watts
         )
-        component_graph = connection_manager.get().component_graph
-        grid_successors = self._get_grid_component_successors()
 
-        if len(grid_successors) == 1:
-            grid_meter = next(iter(grid_successors))
-            if grid_meter.category != ComponentCategory.METER:
-                raise RuntimeError(
-                    "Only grid successor in the component graph is not a meter."
-                )
-            grid_successors = component_graph.successors(grid_meter.component_id)
+        component_graph = connection_manager.get().component_graph
+        # if in the future we support additional producers, we need to add them to the lambda
+        producer_components = component_graph.dfs(
+            self._get_grid_component(),
+            set(),
+            lambda component: component_graph.is_pv_chain(component)
+            or component_graph.is_chp_chain(component),
+        )
 
         first_iteration = True
-        for successor in iter(grid_successors):
-            # if in the future we support additional producers, we need to add them here
-            if component_graph.is_chp_chain(successor) or component_graph.is_pv_chain(
-                successor
-            ):
-                if not first_iteration:
-                    builder.push_oper("+")
+        for component in producer_components:
+            if not first_iteration:
+                builder.push_oper("+")
+            first_iteration = False
```

Comment:
I'd suggest to replace this flag by using the same technique than in pv power formula, by enumerating the components and checking idx greater than 0 to push the operator "+". This is just for consistency